### PR TITLE
feat: Block limits

### PR DESCRIPTION
**Pull request by kendallstrautman**
Pretty simple implementation. The add block and trash icons are hidden / shown based on the limits. Checkout the blocks page in `demo-next` for an example. When the range is limited (i.e. min === max), neither add block or trash icon will show. 

**Example**
```jsx
<InlineBlocks
  name="blocks"
  blocks={PAGE_BUILDER_BLOCKS}
  min= {1}
  max={4}
/>
```

---
- [x] Docs updates PR approved

_Note: I will write docs if we like this API_

### CODE REVIEW

File: packages/react-tinacms-inline/src/blocks/inline-field-blocks.tsx [hunk 1]
```diff
@@ -53,6 +57,10 @@ export interface InlineBlocksActions {
   activeBlock: number | null
   setActiveBlock: any
   direction: 'vertical' | 'horizontal'
+  limits?: {
+    min: number
+    max: number
```
Comment (ncphillips) [inline packages/react-tinacms-inline/src/blocks/inline-field-blocks.tsx:62]:
```suggestion
    min?: number
    max?: number
```

**Comment by ncphillips** [comment]
Hmm looking at it again, instead of `limits` what about just putting `min` and `max` at the root?

```jsx
<InlineBlocks
  name="blocks"
  blocks={PAGE_BUILDER_BLOCKS}
  min={1}
  max={4}
/>
```

**Comment by kendallstrautman** [comment]
good call @ncphillips, that simplified things.

File: packages/demo-next/pages/blocks.tsx [hunk 2]
```diff
@@ -62,6 +62,10 @@ export default function BlocksExample({ jsonFile }) {
               itemProps={{
                 style: { backgroundColor: 'red' },
               }}
+              limits={{
+                min: 2,
+                max: 4,
+              }}
```
Comment (ncphillips) [inline packages/demo-next/pages/blocks.tsx:68]:
```suggestion
              min={2}
              max={4}
```

File: packages/react-tinacms-inline/src/blocks/inline-block-field-controls.tsx [hunk 3]
```diff
@@ -120,6 +123,12 @@ export function BlocksControls({
     .slice(0, -1)
     .join('.')
 
+  function withinLimit(limit: number | undefined) {
+    if (!limit) return true
+
+    return !(limit === count || (max === count && min === count))
+  }
```
Comment (ncphillips) [inline packages/react-tinacms-inline/src/blocks/inline-block-field-controls.tsx:130]: Nice changeup! I like this fn

**Pull request merged**